tools/azsdk-cli/Azure.Sdk.Tools.Cli.Tests/Tools/Generators/ReadMeGeneratorToolTests.cs

```diff
@@ -0,0 +1,162 @@
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using Azure.AI.OpenAI;
+using Azure.Sdk.Tools.Cli.Services;
+using Azure.Sdk.Tools.Cli.Tests.MockServices;
+using Azure.Sdk.Tools.Cli.Tests.TestHelpers;
+using Azure.Sdk.Tools.Cli.Tools;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Moq;
+using OpenAI.Chat;
+
+namespace Azure.Sdk.Tools.Cli.Tests.Tools.Generators
+{
+    internal class ReadMeGeneratorToolTests
+    {
+        [Test]
+        public async Task TestReadmeGeneratorTool()
+        {
+            var testClients = SetupOpenAIMocks();
+            (DirectoryInfo root, string packagePath) = await CreateFakeLanguageRepo();
+
+            var readmeOutputPath = Path.GetTempFileName();
+            var readmeTemplatePath = Path.Combine(AppContext.BaseDirectory, "TestAssets", "README-template.go.md");
+
+            try
+            {
+                var tool = ActivatorUtilities.CreateInstance<ReadMeGeneratorTool>(testClients.ServiceProvider);
+                var command = tool.GetCommand();
+
+                int exitCode = command.Invoke($"--output-path \"{readmeOutputPath}\" --service-url \"https://learn.microsoft.com/azure/service-bus-messaging\" --template-path {readmeTemplatePath} --package-path {packagePath}");
+
+                Assert.Multiple(() =>
+                {
+                    Assert.That(exitCode, Is.EqualTo(0), "Command should execute successfully");
+                    Assert.That(testClients.OutputService.Outputs.First().Method, Is.EqualTo("Output"));
+                    Assert.That(testClients.OutputService.Outputs.First().OutputValue, Is.EqualTo($"Readme written to {readmeOutputPath}"));
+                });
+
+                Assert.That(File.Exists(readmeOutputPath), Is.True, "Readme output file should be created");
+            }
+            finally
+            {
+                Directory.Delete(root.FullName, true);
+            }
+        }
+
+        [Test]
+        public void TestReadmeGeneratorToolLive()
+        {
+            var endpoint = Environment.GetEnvironmentVariable("AZURE_OPENAI_ENDPOINT");
```

Comment:
Fine for now but I'm taking a note to prioritize a higher level configuration class.

Reply:
Yeah, this is a stop-gap measure. An injectable test object would be preferrable.

I do setup the service collection below so that's easy to do.